### PR TITLE
Load synthetic data from the local file system

### PR DESCRIPTION
**Pull request by whytheplatypus**
Provides a loadLocalSyntheticData "test" function and supporting behaviors to load synthatic data from the local filesystem into a local database.

Also allows a local or test database to be named bfddb making it work with bb.dev docker-compose configuration.

### CODE REVIEW

File: apps/bfd-pipeline/bfd-pipeline-rif-load/src/test/java/gov/cms/bfd/pipeline/rif/load/RifLoaderIT.java [hunk 1]
```diff
@@ -131,6 +131,21 @@ public void loadSyntheticData() {
     loadSample(dataSource, StaticRifResourceGroup.SYNTHETIC_DATA);
   }
 
+  /**
+   * Runs {@link gov.cms.bfd.pipeline.rif.load.RifLoader} against the {@link
+   * StaticRifResourceGroup#LOCAL_SYNTHETIC} data.
+   */
+  @Test
+  public void loadLocalSyntheticData() {
```
Comment (RickHawesUSDS) [inline apps/bfd-pipeline/bfd-pipeline-rif-load/src/test/java/gov/cms/bfd/pipeline/rif/load/RifLoaderIT.java]: Should this have an @Ignore attribute? Do we want to run this on every test pass?

Comment (whytheplatypus) [reply apps/bfd-pipeline/bfd-pipeline-rif-load/src/test/java/gov/cms/bfd/pipeline/rif/load/RifLoaderIT.java]: Yup good idea. Is there some better way to do this kind of task aside from a test? Does maven (or java in general) have a best practice for utility type commands?

Comment (RickHawesUSDS) [reply apps/bfd-pipeline/bfd-pipeline-rif-load/src/test/java/gov/cms/bfd/pipeline/rif/load/RifLoaderIT.java]: Not sure about what is the practice, but I ignore is used in this file for this load utilities. Personally, I would love to have a developer command-line utility for this project.

File: apps/bfd-model/bfd-model-rif-samples/src/main/java/gov/cms/bfd/model/rif/samples/StaticRifResource.java [hunk 2]
```diff
@@ -355,6 +475,32 @@ public RifFile toRifFile() {
     };
   }
 
+  /**
+   * @param dataSetLocation the {@link TestDataSetLocation} of the file to get a local copy of
+   * @param fileName the name of the specific file in the specified {@link TestDataSetLocation} to
+   *     get a local copy of, e.g. "beneficiaries.rif"
+   * @return a {@link URL} to a local copy of the specified test data file from S3
+   */
+  private static Supplier<URL> localData(TestDataSetLocation dataSetLocation, String fileName) {
```
Comment (karlmdavis) [inline apps/bfd-model/bfd-model-rif-samples/src/main/java/gov/cms/bfd/model/rif/samples/StaticRifResource.java]: Looks like you're not using the `dataSetLocation` parameter?

I'd remove it.

File: apps/bfd-model/bfd-model-rif-samples/src/main/java/gov/cms/bfd/model/rif/samples/StaticRifResource.java [hunk 3]
```diff
@@ -355,6 +475,32 @@ public RifFile toRifFile() {
...
+    return () -> {
+      // Find the build output `target` directory.
+      Path resourceDir = Paths.get(System.getenv("LOCAL_SYNTHETIC_DATA"));
```
Comment (karlmdavis) [inline apps/bfd-model/bfd-model-rif-samples/src/main/java/gov/cms/bfd/model/rif/samples/StaticRifResource.java]: Same as with the last PR review, could you document that parameter in a `README` somewhere?

File: apps/bfd-model/bfd-model-rif/src/main/java/gov/cms/bfd/model/rif/schema/DatabaseTestHelper.java [hunk 4]
```diff
@@ -93,7 +93,10 @@ public static DataSource getTestDatabaseAfterClean() {
 
     // Try to prevent career-limiting moves.
     String url = getTestDatabaseUrl();
-    if (!url.contains("localhost") && !url.contains("127.0.0.1") && !url.contains("hsqldb:mem")) {
+    if (!url.contains("localhost")
+        && !url.contains("127.0.0.1")
+        && !url.contains("hsqldb:mem")
+        && !url.contains("bfddb")) {
```
Comment (karlmdavis) [inline apps/bfd-model/bfd-model-rif/src/main/java/gov/cms/bfd/model/rif/schema/DatabaseTestHelper.java]: Pretty sure this new last clause shouldn't be here.

File: apps/bfd-pipeline/bfd-pipeline-rif-load/src/test/java/gov/cms/bfd/pipeline/rif/load/RifLoaderIT.java [hunk 5]
```diff
@@ -131,6 +131,21 @@ public void loadSyntheticData() {
...
+    /*Assume.assumeTrue(
+    String.format(
+        "Not enough memory for this test (%s bytes max). Run with '-Xmx5g' or more.",
+        Runtime.getRuntime().maxMemory()),
+    Runtime.getRuntime().maxMemory() >= 4500000000L); */
```
Comment (karlmdavis) [inline apps/bfd-pipeline/bfd-pipeline-rif-load/src/test/java/gov/cms/bfd/pipeline/rif/load/RifLoaderIT.java]: Can you add a comment explaining why this code is commented out, and when one might _un_ comment it?

**Pull request closed without merging**